torch/csrc/dynamo/eval_frame.c

```diff
@@ -727,6 +727,33 @@ static int clear_state(PyObject* module) {
 
 bool is_skip_guard_eval_unsafe = false;
 
+static int c_recursion_limit = -1;
+
+int get_c_recursion_limit() {
+  return c_recursion_limit;
+}
+
+static PyObject* get_c_recursion_limit_py(PyObject* dummy, PyObject* args) {
+  return (PyObject*)PyLong_FromInt32(c_recursion_limit);
```

Comment:
You should use either `int32_t` or `int64_t` and use the appropriate function based on CPython def:

[PyObject](https://docs.python.org/3/c-api/structures.html#c.PyObject) *PyLong_FromInt32(int32_t value)
[PyObject](https://docs.python.org/3/c-api/structures.html#c.PyObject) *PyLong_FromInt64(int64_t value)